### ISSUE TITLE
Python test suite refactoring

### ISSUE THREAD

**Issue by gsidebo** [mitodl/micromasters#850]
We have a few issues with our Micromasters test suite. The main issue is speed: it takes about 2 minutes to complete on a normal development machine. Slow test suites are problematic for many reasons, one of them being disruption to our workflow. A lot of these tests rely on external services where they don't need to. Evaluation of expensive test cases will need to be done on a module-by-module/case-by-case basis. Any defined issues should be given the label for this epic.

Lesser (but noteworthy) issues with the test suite in general:

- Invocation is inflexible. pytest.ini hard-codes the flags to invoke coverage and linting. You cannot, for example, run a single test module or a single test case with no coverage or linting using tox. The flags in pytest.ini are applied regardless of the command line options you include. This can be solved somewhat simply using a conftest.py file and providing an option for 'silent' testing 
- Code repetition. One example: dashboard/api_test.py is 1264 lines long, and a huge amount of it is code that is repeated with small changes. This is a problem for all of the familiar reasons associated with a lack of code DRYness. This also increases the burden of fixing test cases for any logic change we make to the code.

**Comment by aliceriot** [comment]
There is also `pytest_minimal.ini` which I set-up when I first got here, it lets you run the tests without all the coverage and linter nonsense. I haven't been doing python stuff really, so I'm not sure if it's still set up correctly.

**Comment by gsidebo** [comment]
yup, i'm aware of it and i use it currently, but we can do a lot better than forcing developers to make changes to a git-tracked file to run tests with certain parameters, then reverting those changes some time before merging
